doc/modules/semi_supervised.rst

```diff
@@ -17,9 +17,52 @@ labeled points and a large amount of unlabeled points.
 
 .. topic:: Unlabeled entries in `y`
 
-    It is important to assign an identifier to unlabeled points along with the
-    labeled data when training the model with the ``fit`` method. The identifier
-    that this implementation uses is the integer value :math:`-1`.
+   It is important to assign an identifier to unlabeled points along with the
+   labeled data when training the model with the ``fit`` method. The
+   identifier that this implementation uses is the integer value :math:`-1`.
+   Note that for string labels, the dtype of `y` should be object so that it
+   can contain both strings and integers.
+
+.. note::
+
+   Semi-supervised algorithms need to make assumptions about the distribution
+   of the dataset in order to achieve performance gains. See `here
+   <https://en.wikipedia.org/wiki/Semi-supervised_learning#Assumptions_used>`_
+   for more details.
+
+.. _self_training:
+
+Self Training
+=============
+
+This self-training implementation is based on Yarowsky's [1]_ algorithm. Using
+this algorithm, a given supervised classifier can function as a semi-supervised
+classifier, allowing it to learn from unlabeled data.
+
+:class:`SelfTrainingClassifier` can be called with any classifier that
+implements `predict_proba`, passed as the parameter `base_classifier`. In
+each iteration, the `base_classifier` predicts labels for the unlabeled
+samples and adds them to the labeled dataset if the classifier's confidence is
+above the `threshold`. The labels used for the final
+fitting as well as the iteration in which each sample was labeled are available
+as attributes. The optional `max_iter` parameter specifies how many times the
+loop is executed at most.
+
+The `max_iter` parameter may be set to `None`, causing the algorithm to
+iterate until all samples have labels.
+
+.. topic:: Examples
+
+  * :ref:`sphx_glr_auto_examples_semi_supervised_plot_self_training_varying_threshold.py`
+  * :ref:`sphx_glr_auto_examples_semi_supervised_plot_semi_supervised_versus_svm_iris.py`
+
+.. topic:: References
+
+..  [1] David Yarowsky. 1995. Unsupervised word sense disambiguation rivaling
```

Comment:
I think the issue is that this line should be indented, see e.g. https://github.com/scikit-learn/scikit-learn/blame/master/sklearn/datasets/descr/rcv1.rst#L70

I'm not sure about the other warning. I can have a look later today.